smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/KafkaThrottledLatestProcessedCommit.java

```diff
@@ -105,11 +104,50 @@ private OffsetStore getOffsetStore(TopicPartition topicPartition) {
      */
     @Override
     public void partitionsAssigned(Set<TopicPartition> partitions) {
-        offsetStores.clear();
+        stopFlushAndCheckHealthTimer();
+
+        // Remove all handled partitions that are not in the given list of partitions
+        for (TopicPartition partition : new HashSet<>(offsetStores.keySet())) {
```

Comment:
Why is tuning the keySet into a HashSet necessary?

Reply:
Because the key set is a live view, and removing from it may trigger a ConcurrentModificationException.

Reply:
But it's all running in the same Vert.X context, no?

Reply:
It's my understanding the the Vert.X Kafka Client calls revoked and assigned within the context.

Reply:
No, it can come from the `remove` just below. It's always safer to create a copy if you want to remove. Another solution would be to create a temporary `Set`, collect the keys to remove and remove outside of the loop. Both require an additional set, but this variant avoids another loop.

Reply:
Gotcha!